components/utils/highlighter.js

```diff
@@ -0,0 +1,13 @@
+export default highlighter
+
+function highlighter(highlight) {
```

Comment:
I was trying a functional approach, but may have gone overboard. ;-) Even with the inner functions being named to, hopefully, represent what they are, the further you get into the layers, the less obvious it is what each function represents.

It leads to something like:
Create a highlighter library specific factory that can create pragma specific highlighting functionality that can handle a specific value/component combination (`code handler`). Whew, what a mouthful!

We could be less functional, reducing the number nested functions and combine some of the arguments into smaller number of functions. There may be other things to do also, what do you think?